spec/stud/interval_spec.rb

```diff
@@ -24,6 +35,59 @@
       end
     end
 
+    it 'should be able to interrupt an interval within same thread' do
+      counter = 0
+
+      Stud.interval(0.5) do
+        counter += 1
```

Comment:
have you considered implementing the counters in these specs with a mock object and the time measurement with Benchmark? e.g.:

```ruby
 it 'should be able to interrupt an interval defined as a task while sleeping' do
   let(:target) { double("target_object") }
   allow(target).to receive(:touch)

   elapsed_time = Benchmark.realtime do
     task = Stud::Task.new do
       Stud.interval(10, :sleep_then_run => true) { target.touch }
     end
     sleep(1)
     task.stop!
     task.wait
   end

   # the execution time worst case should be the sleep(1) call plus a sleep(1) in
   # the interval loop before seeing the stop? and maybe a "reminder" sleep(< 1)
   # so < 3 should be safe.
   expect(elapsed_time).to be < 3
   expect(counter).to eq(0)
end
```